lib/wasi/src/syscalls/wasi/fd_filestat_get.rs

```diff
@@ -26,22 +34,51 @@ pub fn fd_filestat_get<M: MemorySize>(
 /// Output:
 /// - `__wasi_filestat_t *buf`
 ///     Where the metadata from `fd` will be written
-pub(crate) fn fd_filestat_get_internal<M: MemorySize>(
+pub(crate) fn fd_filestat_get_internal(
     ctx: &mut FunctionEnvMut<'_, WasiEnv>,
     fd: WasiFd,
-    buf: WasmPtr<Filestat, M>,
-) -> Errno {
+) -> Result<Filestat, Errno> {
     let env = ctx.data();
-    let (memory, mut state, inodes) = env.get_memory_and_wasi_state_and_inodes(&ctx, 0);
-    let fd_entry = wasi_try!(state.fs.get_fd(fd));
+    let (_, mut state, inodes) = env.get_memory_and_wasi_state_and_inodes(&ctx, 0);
+    let fd_entry = state.fs.get_fd(fd)?;
     if !fd_entry.rights.contains(Rights::FD_FILESTAT_GET) {
-        return Errno::Access;
+        return Err(Errno::Access);
     }
 
-    let stat = wasi_try!(state.fs.filestat_fd(fd));
+    state.fs.filestat_fd(fd)
+}
+
+/// ### `fd_filestat_get_old()`
+/// Get the metadata of an open file
+/// Input:
+/// - `Fd fd`
+///     The open file descriptor whose metadata will be read
+/// Output:
+/// - `Snapshot0Filestat *buf`
+///     Where the metadata from `fd` will be written
+#[instrument(level = "debug", skip_all, fields(%fd), ret)]
+pub fn fd_filestat_get_old<M: MemorySize>(
```

Comment:
I'd put the `_old` functions in `snapshot0.rs` and instead make the `_internal` functions `pub(in crate::syscalls)` instead. 

To prevent confusion.

Reply:
I have left them in the same file because they shared the same logic, and I'm afraid we might get desynchronized if we separate the "_old" one from the new one